easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1152,18 +1153,110 @@ def _validate(self, attr, values):  # private method
         handles metadata for external module dependencies
         """
         dependency = {}
+        dep_name_no_version = dep_name.split('/')[0]
+        metadata_fields = ['name', 'version', 'prefix']
+        external_metadata = {}
+
         if dep_name in self.external_modules_metadata:
-            dependency['external_module_metadata'] = self.external_modules_metadata[dep_name]
-            self.log.info("Updated dependency info with available metadata for external module %s: %s",
-                          dep_name, dependency['external_module_metadata'])
+            external_metadata = self.external_modules_metadata[dep_name]
+            if not all(d in external_metadata for d in metadata_fields):
+                external_metadata = self._handle_ext_module_metadata_by_probing_modules(dep_name,
+                                                                                        dependency=external_metadata)
+            if external_metadata:
+                self.log.info("Updated dependency info with metadata from available modules for external module %s: %s",
+                              dep_name, external_metadata)
+                dependency['external_module_metadata'] = external_metadata
+            else:
+                self.log.info("No metadata available for external module %s.", dep_name)
+        elif dep_name_no_version in self.external_modules_metadata:
+            external_metadata = self.external_modules_metadata[dep_name_no_version]
+            if not all(d in external_metadata for d in metadata_fields):
+                external_metadata = self._handle_ext_module_metadata_by_probing_modules(dep_name_no_version,
+                                                                                        dependency=external_metadata)
+                if external_metadata:
+                    self.log.info("Updated dependency info with metadata from available modules for external module "
+                                  "%s: %s", dep_name, external_metadata)
+                    dependency['external_module_metadata'] = external_metadata
+                else:
+                    self.log.info("No metadata available for external module %s.", dep_name)
         else:
-            self.log.info("No metadata available for external module %s", dep_name)
+            self.log.info("No metadata available for external module %s. Attempting to read from available modules",
+                          dep_name)
+            external_metadata = self._handle_ext_module_metadata_by_probing_modules(dep_name)
+            if external_metadata:
+                dependency['external_module_metadata'] = external_metadata
+                self.log.info("Updated dependency info with metadata from available modules for external module %s: %s",
+                              dep_name, external_metadata)
+            else:
+                self.log.info("No metadata available for external module %s.", dep_name)
```

Comment:
what about simplifying a bit to something like

```python
        if dep_name in self.external_modules_metadata:
            external_metadata = self.external_modules_metadata[dep_name]
        elif dep_name_no_version in self.external_modules_metadata:
            external_metadata = self.external_modules_metadata[dep_name_no_version]

        if external_metadata:
            if all(d in external_metadata for d in metadata_fields):
                dependency['external_module_metadata'] = external_metadata
            else:
                self.log.info("Not all metadata available for external module %s. Attempting to read from available modules",
                              dep_name)
                external_metadata = self._handle_ext_module_metadata_by_probing_modules(dep_name,
                                                                                        dependency=external_metadata)
                if external_metadata:
                    dependency['external_module_metadata'] = external_metadata
                    self.log.info("Updated dependency info with metadata from available modules for external module %s: %s",
                                  dep_name, external_metadata)
                else:
                    self.log.info("No metadata available for external module %s.", dep_name)
        else:
            self.log.info("No metadata available for external module %s. Attempting to read from available modules",
                          dep_name)
            external_metadata = self._handle_ext_module_metadata_by_probing_modules(dep_name)
            if external_metadata:
                dependency['external_module_metadata'] = external_metadata
                self.log.info("Updated dependency info with metadata from available modules for external module %s: %s",
                              dep_name, external_metadata)
            else:
                self.log.info("No metadata available for external module %s.", dep_name)
```

Reply:
Fully agreed, it's way too copy-pasty right now.
The `if` and `elif` are exactly the same, except for `dep_name` vs `dep_name_no_version`...

We can do better, I'm looking into it.